packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx

```diff
@@ -116,23 +117,43 @@ export const MoreDrawer = (props: MoreDrawerProps): JSX.Element => {
   );
 
   if (props.microphones && props.microphones.length > 0) {
-    drawerMenuItems.push({
-      itemKey: 'microphones',
-      text: props.strings.microphoneMenuTitle,
-      iconProps: { iconName: 'MoreDrawerMicrophones' },
-      subMenuProps: props.microphones.map((mic) => ({
-        itemKey: mic.id,
-        iconProps: {
-          iconName: isDeviceSelected(mic, props.selectedMicrophone)
-            ? 'MoreDrawerSelectedMicrophone'
-            : 'MoreDrawerMicrophones'
-        },
-        text: mic.name,
-        onItemClick: onMicrophoneItemClick,
-        secondaryIconProps: isDeviceSelected(mic, props.selectedMicrophone) ? { iconName: 'Accept' } : undefined
-      })),
-      secondaryText: props.selectedMicrophone?.name
-    });
+    if (props.speakers && props.speakers.length > 0) {
```

Comment:
nit: a lot of the props are the same for the drawer menu item, consider pulling out just the props that changed based on when speakers aren't present. Something like:
```
// Add comment also when and why we use the generic audioDevices over Microphone also
const speakersAvailable = props.speakers && props.speakers.length > 0;
const itemKey = speakersAvailable ? 'microphones' : 'audioDevices';
const text = speakersAvailable ? etc.

```

Reply:
Made some props for those cases.